core/src/main/java/org/elasticsearch/common/unit/Fuzziness.java

```diff
@@ -103,10 +114,28 @@ public static Fuzziness build(Object fuzziness) {
         String string = fuzziness.toString();
         if (AUTO.asString().equalsIgnoreCase(string)) {
             return AUTO;
+        }  else if (string.toUpperCase(Locale.ROOT).startsWith(AUTO.asString() + ":")) {
+            return parseCustomAuto(string);
         }
         return new Fuzziness(string);
     }
 
+    private static Fuzziness parseCustomAuto( final String string) {
+        String[] fuzzinessLimit = string.substring(AUTO.asString().length() + 1).split(",");
```

Comment:
can you add an `assert string.toUpperCase(Locale.ROOT).startsWith(AUTO.asString() + ":")` on top of that line? Just to make sure tests would catch if we mistakenly start to call this method without performing that check first